src/server/tests/test_dal.py

```diff
@@ -32,39 +33,90 @@ def test_get_user_with_unknown_user_should_return_none(self):
+
+    def test_save_then_get_user_docs_should_be_equals(self):
+        # setup : init docs in database
         db_doc1 = db.Document.make(url='url1', title='title1', summary=None)
         db_doc2 = db.Document.make(url='url2', title='title2', summary=None)
+        ndb.put_multi([db_doc1, db_doc2])
 
-        feature1 = db.FeatureDescription.make('desc1')
-        feature2 = db.FeatureDescription.make('desc2')
-        feature_set = db.FeatureSet.make(feature_set_id='set', feature_descriptions=[feature1, feature2])
-        ndb.put_multi([db_doc1, db_doc2, feature_set])  # we must store entities before creating user to init key fields
+        doc1 = struct.Document.make_from_db(
+            url=db_doc1.url, title=db_doc1.title, summary=None, date=None, db_key=db_doc1.key)
+        doc2 = struct.Document.make_from_db(
+            url=db_doc2.url, title=db_doc2.title, summary=None, date=None, db_key=db_doc2.key)
 
-        doc1 = dal.Document(url=db_doc1.url, title=db_doc1.title, db_key=db_doc1.key)
-        doc2 = dal.Document(url=db_doc2.url, title=db_doc2.title, db_key=db_doc2.key)
+        expected_user_docs = [struct.UserDocument(doc1, 0.1), struct.UserDocument(doc2, 0.2)]
 
-        feature_vector = dal.FeatureVector(vector=[0.5, 0.6], labels=['desc1', 'desc2'], feature_set_id=feature_set.key.id())
+        # create user and save it to init user_document_set_key field
+        user = struct.User.make_from_scratch("test_save_then_get_user_docs_should_be_equals")
+        dal.save_user(user)
 
-        user_docs = [dal.UserDocument(doc1, 0.1), dal.UserDocument(doc2, 0.2)]
-        expected_user = dal.User(email='email', user_docs=user_docs, feature_vector=feature_vector)
+        dal.save_user_docs(user, expected_user_docs)
+        result_user_docs = dal.get_user_docs(user)
+        self.assertEquals(len(expected_user_docs), len(result_user_docs))
 
-        # ------------- Save then get user --------------
-        dal.save_user(expected_user)
-        result_user = dal.get_user('email')
+        for (expected, result) in itertools.izip_longest(expected_user_docs, result_user_docs):
+            self.assertEquals(expected.grade, result.grade)
+            self.assertEquals(expected.document.title, result.document.title)
 
-        # --------------assert equality ------------------
-        self.assertEquals(expected_user.email, result_user.email)
+    def test_save_documents(self):
+        expected_doc1 = struct.Document.make_from_scratch(
+            url='url1_test_save_documents', title='title1_test_save_documents', summary='summary1')
+        expected_doc2 = struct.Document.make_from_scratch(
+            url='url2_test_save_documents', title='title1_test_save_documents', summary='summary2')
+        expected_docs = [expected_doc1, expected_doc2]
+        dal.save_documents(expected_docs)
+        for expected_doc in expected_docs:
+            result_doc = expected_doc.db_key.get()
```

Comment:
The attribute db_key indirectly exposes "database implementation" ? By calling db_key.get() we can get a db object ? If yes, we should only deal with objects of "frontendstructs.py" (our model).